File: ABC/C/257/C.py
n = int(input())
ans = 0
vvec = []
s = input()
w = list(map(int, input().split()))
for i in range(n):
    vvec.append([w[i], s[i]])
    if s[i] == '1': ans += 1
vvec = sorted(vvec)
cnt = ans
for i in range(n):
    if vvec[i][1] == '1': cnt -= 1
    else: cnt += 1
    if i < n - 1:
        if vvec[i][0] != vvec[i+1][0]:
            ans = max(ans, cnt)
    else: 
        ans = max(ans, cnt)
print(ans)

File: ABC/C/257/C.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main(){
    int n, ans=0, cnt=0;
    string s;
    cin >> n >> s;
    vector<pair<int, char>> pvec;
    for(int i=0; i<n; i++){
        int w;
        cin >> w;
        //pvec.push_back(make_pair(w, s.at(i)));
        if(s.at(i) == '1') ans++;
    }
    sort(pvec.begin(), pvec.end());
    cnt = ans;
    for(int i=0; i<n; i++){
        if(pvec.at(i).second == '1') cnt--;
        else cnt++;
        if(i < n - 1){
            if(pvec.at(i).first != pvec.at(i+1).first) ans = max(ans, cnt);
        }
        else{
            ans = max(ans, cnt);
        }
    }
    cout << ans << endl;
}